src/Controllers/DatabaseController.js

```diff
@@ -504,7 +504,9 @@ DatabaseController.prototype.addInObjectIdsIds = function(ids = null, query) {
     idsIntersection = intersect(allIds);
   }
 
-  // Need to make sure we don't clobber existing $lt or other constraints on objectId
+  // Need to make sure we don't clobber existing $lt or other constraints on objectId.
+  // Clobbering $eq, $in and shorthand $eq (query.objectId === 'string') constraints
+  // is expected though.
```

Comment:
@drew-gross @flovilmart 
I ran into an issue migrating an app to Parse Server and I'm hoping you can shed some light on this. You can see the [failing test I wrote here](https://github.com/JeremyPlease/parse-server/blob/notequalto-query-relations/spec/ParseQuery.spec.js#L121-L129). On api.parse.com and Parse Server <= 2.2.4 this test passes.

I tracked the problem down to this comment. Should we really be clobbering $eq constraints here or should the test linked above pass? The hacky feeling workaround would be to replace `query.equalTo("objectId", cake1.id)` with `query.containedIn("objectId", [cake1.id])` in my test.

Thanks!